### PR TITLE
Refactored lab tests

### PR DESCRIPTION
**Pull request by fredkingham**

### CODE REVIEW

File: plugins/labtests/models.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    return dt_format(some_dt, settings.DATETIME_FORMAT)
+
+
+"""
+Models for labtests
+"""
+class LabTest(models.Model):
+    patient = models.ForeignKey(
+        omodels.Patient,
+        on_delete=models.CASCADE,
```
Comment (davidmiller) [inline plugins/labtests/models.py:20]: Does this definitely work the way we want it to ?

File: plugins/labtests/models.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    status = models.CharField(max_length=256, blank=True, null=True)
+    test_code = models.CharField(max_length=256, blank=True, null=True)
+    test_name = models.CharField(max_length=256, blank=True, null=True)
+    lab_number = models.CharField(max_length=256, blank=True, null=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    @classmethod
+    def create_from_old_test(cls, lt):
+        s = cls()
```
Comment (davidmiller) [inline plugins/labtests/models.py]: lots of single name variables here...

File: plugins/labtests/models.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        fields = [
+            "clinical_info",
+            "site",
+            "status",
+            "test_code",
+            "test_name",
+            "lab_number",
+        ]
+        for f in fields:
+            setattr(s, f, extras.get(f, None))
```
Comment (davidmiller) [inline plugins/labtests/models.py:47]: I'd swear a lot if this line ever turned up in a stacktrace :)

File: plugins/labtests/models.py [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+            "observation_value",
+            "reference_range",
+            "units"
+        ]
+        for field in fields:
+            as_dict[field] = getattr(self, field)
+        return as_dict
+
+
+def create_from_old_tests(patient):
```
Comment (davidmiller) [inline plugins/labtests/models.py]: I think this whole thing could be lost and make the management command iterate over upstream tests?

File: elcid/api.py [hunk 5]
```diff
@@ -78,6 +80,26 @@
         LAB_TEST_TAGS[test_name].append(tag)
 
 
+def get_upstream_lab_tests_for_patient(user, patient):
```
Comment (davidmiller) [inline elcid/api.py]: none of these three functions are tested...

File: plugins/labtests/models.py (hunk 2 above)
Comment (fredkingham) [reply plugins/labtests/models.py]: resolved on 9791e5ea

File: plugins/labtests/models.py (hunk 3 above)
Comment (fredkingham) [reply plugins/labtests/models.py:47]: resolved on 83b691d6

File: plugins/labtests/models.py (hunk 4 above)
Comment (fredkingham) [reply plugins/labtests/models.py]: resolved on 83b691d6

File: elcid/api.py (hunk 5 above)
Comment (fredkingham) [reply elcid/api.py]: resolved on 781046f6

**Comment by fredkingham** [comment]
@davidmiller I think these are all resolved. Thoughts?

**Pull request merged**